spec/features/staff/users_can_edit_an_activity_spec.rb

```diff
@@ -413,6 +413,42 @@
         click_button t("form.button.activity.submit")
         expect(page).to have_content t("activerecord.errors.models.activity.attributes.planned_start_date.invalid")
       end
+
+      it "the policy markers can be added" do
+        activity = create(:project_activity, organisation: user.organisation)
+        # Report needs to exist so the activity is editable
+        _report = create(:report, state: :active, organisation: user.organisation, fund: activity.associated_fund)
+
+        visit organisation_activity_details_path(activity.organisation, activity)
+
+        within(".policy_marker_gender") do
+          click_on(t("default.link.edit"))
+          expect(page).to have_current_path(
+            activity_step_path(activity, :policy_markers)
+          )
+        end
+
+        choose "activity-policy-marker-gender-significant-objective-field"
+        click_button t("form.button.activity.submit")
+
+        activity.reload
+        expect(activity.policy_marker_gender).to eql("significant_objective")
```

Comment:
These feature specs seem to highlight the problem behaviour perfectly and should prevent regressions.

Just a small suggestion here: as this is an integration test I think it would be ideal to keep our assertions to what the user sees on screen. i.e. rather than reloading the activity and making an assertion on value of the `#policy_market_gender` attribute, we could assert that we see the expected "Significant objective" label on the screen, e.g. with

```ruby
expect(page).to have_css(".policy_marker_gender dd", text: "Significant objective")
```

What do you think?

Reply:
I think it's great! I'm just surprised that it works, because the other week I lost half a day trying to debug a `StaleElementReferenceError` when I tried to test both scenarios in one.

Capybara didn't seem to like `page` being re-used after going into editing mode again - that's why I split them into two scenarios in the end. 🤯 

I thought the reference went stale because of executing another request, but it turns out I don't really understand what makes it stale and what doesn't.